src/PlatformSwitch/hooks.ts

```diff
@@ -0,0 +1,57 @@
+import { redirect } from '@graasp/sdk';
+
+/** Enumeration of available platforms */
+export enum Platform {
+  Builder = 'Builder',
+  Player = 'Player',
+  Library = 'Library',
+  Analytics = 'Analytics',
+}
+
+/** Maps each Platform to a URL generator function */
+export type HostsMapper = Partial<Record<Platform, (itemId: string) => string>>;
+
+/**
+ * Generates a default hosts mapper given a record of platform to hostname
+ * Only the origin part of the given hostname will be used!
+ *
+ * The default mapping is:
+ *  BUILDER_HOST/items/<itemId>
+ *  PLAYER_HOST/<itemId>
+ *  LIBRARY_HOST/<itemId>
+ *
+ * For any advanced usage, create your own {@see HostsMapper}
+ */
+export function defaultHostsMapper(
+  hostsUrls: Partial<Record<Platform, string>>,
+): HostsMapper {
+  return Object.fromEntries(
+    Object.entries(hostsUrls).map(([platform, url]) => {
+      const origin = new URL(url).origin;
+      // in Builder platform, should appear under /items
+      const prefixIfBuilder = platform === Platform.Builder ? 'items/' : '';
+      return [
+        platform,
+        (itemId: string) => `${origin}/${prefixIfBuilder}${itemId}`,
+      ];
+    }),
```

Comment:
Actually there's util function to build a builder link https://github.com/graasp/graasp-sdk/blob/80bd60fb83e2b5fe3721a975184f99265b1dae66/src/utils/navigation.ts#L81
I didn't know where to set the responsibility of defining the url. But since we used it at many places so it was simpler to reuse this bit of code.

Reply:
I think we have a weird situation where each platform knows best what link is relevant for a given item ID, but another platform may want to use a specific link for a given functionality of another platform. Also in this case, we should be using `buildItemLinkForBuilder` only if the platform is Builder, otherwise the route is different. 

What about having platforms expose some kind of "default" route for any itemId, that would show the relevant "home" page for that id? Not sure where we should put this either.